lib/src/facts/aix/load_average_resolver.cc

```diff
@@ -0,0 +1,51 @@
+#include <internal/facts/aix/load_average_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/locale/locale.hpp>
+
+#include <sys/inttypes.h>
+#include <sys/kinfo.h>
+
+// Mark string for translation (alias for leatherman::locale::format)
+using leatherman::locale::_;
+
+using namespace std;
+
+/**
+ * This system call lets us query the kernel directly for system
+ * information. We use it to get our current load averages.
+ *
+ * @param info the info we're retrieving from the kernel.
+ * @param buf the buffer that we'll store the information in
+ * @param buf_size a pointer to the variable containing the size of the buffer in bytes
+ * @param arg no idea what this param. represents. we will usually set this to 0.
+ * @return 0 if we successfully retrieve the information, else a negative value
+ */
+extern "C" int getkerninfo(int info, char* buf, int* buf_size, int32long64_t arg);
```

Comment:
I see this declaration in https://github.com/puppetlabs/facter/blob/3.11.x/lib/src/facts/aix/networking_resolver.cc#L34. Should we move it over to a separate .h file? E.g. something like `util/aix/getkerninfo.hpp`? Also, I tried using https://github.com/puppetlabs/facter/blob/3.11.x/lib/src/facts/aix/networking_resolver.cc#L37 here but it throws a bad alloc exception b/c `getkerninfo(KINFO_GET_AVENRUN, nullptr, nullptr, 0)` returns -1. Probably b/c there is no load averages struct from what I could see in the IBM header files while the networking stuff has their own structs.

Reply:
I think your usage below is fine, since there's no struct for this KINFO value.

Regarding moving it to a header, I don't think we have a good place to put it right now? Seems silly to add a new header for one declaration